proxy/proxy.go

```diff
@@ -555,13 +567,29 @@ func (p *Proxy) checkBreaker(c *context) (func(bool), bool) {
 	return done, ok
 }
 
+func ratelimitError(settings ratelimit.Settings) error {
+	return &proxyError{
+		err:  errRatelimitError,
+		code: http.StatusTooManyRequests,
+		additionalHeader: http.Header{
+			ratelimit.Header: []string{strconv.Itoa(settings.MaxHits * int(time.Hour/settings.TimeWindow))},
+		},
+	}
+}
+
 func (p *Proxy) do(ctx *context) error {
 	if ctx.loopCounter > p.maxLoops {
 		return errMaxLoopbacksReached
 	}
 
 	ctx.loopCounter++
 
+	if ok, settings := p.limiters.Check(ctx.request); !ok {
```

Comment:
nit: it's normal in go to have the variable order switched like: `settings, ok := ..`